starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -9,27 +11,38 @@
 
 import ly.generalassemb.drewmahrt.shoppinglistdetailview.setup.DBAssetHelper;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements ListFragment.OnItemSelectedListener {
+
+    public static final String NAME_KEY = "name";
+    public static final String DESCRIPTION_KEY = "description";
+    public static final String PRICE_KEY = "price";
+    public static final String TYPE_KEY = "type";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        //Ignore the two lines below, they are for setup
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
-        //Setup the RecyclerView
-        RecyclerView shoppingListRecyclerView = (RecyclerView) findViewById(R.id.shopping_list_recyclerview);
-
-        ShoppingSQLiteOpenHelper db = ShoppingSQLiteOpenHelper.getInstance(this);
-        List<ShoppingItem> shoppingList = db.getShoppingList();
-
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false);
-
-        shoppingListRecyclerView.setLayoutManager(linearLayoutManager);
-        shoppingListRecyclerView.setAdapter(new ShoppingListAdapter(shoppingList));
+        Fragment fragment = ListFragment.newInstance(null, this);
+        getSupportFragmentManager().beginTransaction()
+                .add(R.id.fragment_container, fragment).commit();
+    }
 
+    @Override
+    public void onItemSelected(ShoppingItem selected) {
+        Bundle bundle = new Bundle();
+        bundle.putString(NAME_KEY, selected.getName());
+        bundle.putString(DESCRIPTION_KEY, selected.getDescription());
+        bundle.putString(PRICE_KEY, selected.getPrice());
+        bundle.putString(TYPE_KEY, selected.getType());
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you're not passing these values via intent, but it's almost the same thing. This works OK in this case, but what if the item had 100 fields of data? Then you'd want to pass just the item ID here, and make a separate database query from the detail fragment to get the one item you need.